File: solutions/5/sol.py
def gcd(a, b):
  if a==0:
    return b
  if b==0:
    return a
  if a>b:
    return gcd(a-b, b)
  else:
    return gcd(b-a, a)

print(gcd(12, 16))
print(gcd(9, 28))

File: solutions/5/sol.go
package main

import (
	"fmt"
)

// Greatest common divisor
func gcd(a int, b int) int {
	if a == 0 {
		return b
	}
	if a > b {
		return gcd(a%b, b)
	} else {
		return gcd(b%a, a)
	}
}

// Least common multiple
// lcm

func main() {
	fmt.Print(gcd(10, 12))
}
